### PR TITLE
chore: add bounded sdp declaration handling to genesis block

### PR DESCRIPTION
**Pull request by hansieodendaal**
## 1. What does this PR implement?

Added bounded SDP declaration handling to genesis block construction.
- Reserve capacity for the required transfer and inscription operations.
- Add checked collection, single-item push, and batch-extension helpers for `GenesisSDPDeclareOps`.
- Return `TooManyOps` errors instead of silently overflowing or panicking.
- Make bulk declaration builder methods fallible.
- Add boundary tests for maximum capacity and overflow cases.

## 2. Does the code have enough context to be clearly understood?

Yes

## 3. Who are the specification authors and who is accountable for this PR?

@hansieodendaal 

## 4. Is the specification accurate and complete?

Yes

## 5. Does the implementation introduce changes in the specification?

No

## Checklist

> [!WARNING]  
> Do not merge the PR if any of the following is missing:

* [X] 1. The PR title follows the Conventional Commits [specification](https://www.notion.so/How-to-open-a-Pull-Request-215261aa09df80deb538ed59f5e4e0b5).
* [X] 2. Description added.
* [X] 3. Context and links to Specification document(s) added.
* [X] 4. Main contact(s) (developers and specification authors) added
* [X] 5. Implementation and Specification are 100% in sync including changes. This is critical.
* [X] 6. Link PR to a specific milestone.
* [X] 7. New or changed logs were reviewed against the [logging guidelines](https://github.com/logos-blockchain/logos-blockchain/blob/master/CONTRIBUTING.md#logging-guidelines).

### CODE REVIEW

File: core/src/block/genesis.rs [hunk 1]
```diff
@@ -801,13 +784,74 @@ impl GenesisBlockBuilder<WithNotesAndInscription> {
...
+    for item in new_items {
+        declarations.try_push(item).map_err(|_| {
+            Error::InvalidGenesisTx(genesis_tx::Error::TooManyOps {
+                count: GENESIS_REQUIRED_OPS + declarations.len() + 1,
+            })
+        })?;
+    }
+    Ok(declarations)
+}
+
```
Comment (danielSanchezQ) [inline core/src/block/genesis.rs:858]: Why are these not methods of the `GenesisSDPDeclareOps`?`

**Review by danielSanchezQ — APPROVED** [review]

File: core/src/block/genesis.rs (hunk 1 above)
Comment (hansieodendaal) [reply core/src/block/genesis.rs:858]: `GenesisSDPDeclareOps` is only a type alias:
```rust
pub type GenesisSDPDeclareOps =
    UpperBoundedVec<SDPDeclareOp, MAX_GENESIS_DECLARATIONS>;
```
and `impl GenesisSDPDeclareOps` is not allowed (`Inherent `impl` cannot be defined for a type outside its containing crate [E0116]`).

However, we could implement an extension trait `trait GenesisSdpDeclareOpsExt` and implement it for the alias as `impl GenesisSdpDeclareOpsExt for GenesisSDPDeclareOps {` for some of the pertinent helpers above, but I think that would be overly complex.

Comment (danielSanchezQ) [reply core/src/block/genesis.rs:858]: yeah, not worth it. thanks.
